Bottom: ca11630e9238f84d73e151cd9ee145a342f16329
Top:    1203552e0323bbff82207498c9c26c1c3f05f56b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-23 17:20:19 +0000

report.c: Add paranoid undo serial increment in Report[All]NetLength[s]

This is unlikely to be a problem unless a conneciton is not found.


---

diff --git a/src/report.c b/src/report.c
index 44cba19..07ff6b8 100644
--- a/src/report.c
+++ b/src/report.c
@@ -689,7 +689,9 @@ ReportAllNetLengths (int argc, char **argv, int x, int y)
 
     got_one:
       ResetConnections (true);
-      /* XXX: Might need a serial number increment?? */
+      /* NB: XYtoNetLength calls LookupConnection, which performs an undo
+       *     serial number update, so we don't need to add one here.
+       */
       length = XYtoNetLength (x, y, &found);
 
       gui->log("Net %s length %.*f %s\n", netname, prec, length*scale, units_name);
@@ -705,7 +707,9 @@ ReportNetLength (int argc, char **argv, int x, int y)
   int found = 0;
 
   ResetConnections (true);
-  /* XXX: Might need a serial number increment?? */
+  /* NB: XYtoNetLength calls LookupConnection, which performs an undo
+   *     serial number update, so we don't need to add one here.
+   */
   gui->get_coords ("Click on a connection", &x, &y);
 
   length = XYtoNetLength (x, y, &found);
